src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/EnvelopeReferenceAppender.java

```diff
@@ -0,0 +1,91 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.config.ServiceConfigItem;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.config.ServiceConfiguration;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.model.ccd.CaseAction;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.model.ccd.CcdCollectionElement;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.model.ccd.EnvelopeReference;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static java.util.Arrays.asList;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * Modifies case's bulkScanEnvelopes field by (conditionally) appending envelope information to existing items.
+ */
+@Service
+public class EnvelopeReferenceAppender {
+
+    private final ObjectMapper objectMapper;
+    private final Set<String> enabledServices;
+
+    public EnvelopeReferenceAppender(
+        ObjectMapper objectMapper,
+        ServiceConfiguration serviceConfiguration
```

Comment:
`@Service` annotation should have `@EnableConfigurationProperties`, no matter if it was already present somewhere but never trust a lazy autoload of the spring boot :wink:

Reply:
Good point, now I realise I should have used ServiceConfigProvider to retrieve those.

Reply:
done - replaced with ServiceConfigProvider